lib/logstash/codecs/auto_flush.rb

```diff
@@ -11,16 +11,21 @@ def start
     # can't start if pipeline is stopping
     return self if stopped?
     if pending?
-      @task.reset
+      @task.cancel
```

Comment:
`ScheduledTask#cancel` will only cancel class that are `pending`

from https://github.com/ruby-concurrency/concurrent-ruby/blob/3d6bdecb0c02128656750dfdfa7f21562f408949/lib/concurrent/scheduled_task.rb#L42-L45
```
# A `:pending` task can be cancelled using the `#cancel` method. A task in any
# other state, including `:processing`, cannot be cancelled. The `#cancel`
# method returns a boolean indicating the success of the cancellation attempt.
# A cancelled `ScheduledTask` cannot be restarted. It is immutable.
```

So if the task is currently processing something, this wont cancel the task and create a new one?
Might be good to check if the cancel call was successful, if its the case we start a new task?